python/pyarrow/tests/test_compute.py

```diff
@@ -1854,6 +1896,11 @@ def test_count():
     assert pc.count(arr, mode='only_valid').as_py() == 3
     assert pc.count(arr, mode='only_null').as_py() == 2
     assert pc.count(arr, mode='all').as_py() == 5
+    assert pc.count(arr, 'all').as_py() == 5
```

Comment:
I am wondering here, should we rather make the `mode` keyword keyword-only now we are changing this? 
I don't find it super clear what the "all" in `pc.count(arr, 'all')` would mean.

Reply:
Actually, it's not that "mode" is that a clarifying keyword name either .., so ignore my comment.

Reply:
Yes, allowing positional arguments is always a balancing act...